### DISCUSSION TITLE
URLError: <urlopen error [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1131)>

### DISCUSSION THREAD

**Discussion by SuperFDX** [obss/sahi#759] [Q&A]
Output exceeds the [size limit](command:workbench.action.openSettings?[). Open the full output data [in a text editor](command:workbench.action.openLargeOutput?39d3d283-87a5-40f9-a3aa-d5502d3e1b20)
---------------------------------------------------------------------------
SSLError                                  Traceback (most recent call last)
File ~/anaconda3/envs/sahi/lib/python3.8/urllib/request.py:1354, in AbstractHTTPHandler.do_open(self, http_class, req, **http_conn_args)
   1353 try:
-> 1354     h.request(req.get_method(), req.selector, req.data, headers,
   1355               encode_chunked=req.has_header('Transfer-encoding'))
   1356 except OSError as err: # timeout error

File ~/anaconda3/envs/sahi/lib/python3.8/http/client.py:1256, in HTTPConnection.request(self, method, url, body, headers, encode_chunked)
   1255 """Send a complete request to the server."""
-> 1256 self._send_request(method, url, body, headers, encode_chunked)

File ~/anaconda3/envs/sahi/lib/python3.8/http/client.py:1302, in HTTPConnection._send_request(self, method, url, body, headers, encode_chunked)
   1301     body = _encode(body, 'body')
-> 1302 self.endheaders(body, encode_chunked=encode_chunked)

File ~/anaconda3/envs/sahi/lib/python3.8/http/client.py:1251, in HTTPConnection.endheaders(self, message_body, encode_chunked)
   1250     raise CannotSendHeader()
-> 1251 self._send_output(message_body, encode_chunked=encode_chunked)

File ~/anaconda3/envs/sahi/lib/python3.8/http/client.py:1011, in HTTPConnection._send_output(self, message_body, encode_chunked)
   1010 del self._buffer[:]
-> 1011 self.send(msg)
   1013 if message_body is not None:
   1014 
...
-> 1357         raise URLError(err)
   1358     r = h.getresponse()
   1359 except:

URLError: <urlopen error [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1131)>

I ran the code in this file（inference_for_mmdetection.ipynb） and reported the above error. How can I solve it? Thank you!
     model_path = 'models/cascade_mask_rcnn.pth'
     download_mmdet_cascade_mask_rcnn_model(model_path)
     config_path = download_mmdet_config(model_name="cascade_rcnn", config_file_name="cascade_mask_rcnn_r50_fpn_1x_coco.py",)
     download_from_url('https://raw.githubusercontent.com/obss/sahi/main/demo/demo_data/small-vehicles1.jpeg', 'demo_data/small- 
     vehicles1.jpeg')
     download_from_url('https://raw.githubusercontent.com/obss/sahi/main/demo/demo_data/terrain2.png', 'demo_data/terrain2.png')

**Comment by fcakyon** [comment]
It means you are not able to download the images and models, try to download them manually or check your internet connection :)

**Reply by SuperFDX** [reply]
Thank you for your reply, but I think the source code I downloaded already contains images and models. How can I modify the code so that it does not need to download images and models? Looking forward to your reply, thank you.
